mdsuite/file_io/extxyz_files.py

```diff
@@ -133,7 +133,13 @@ def _get_metadata(self):
                 self.tabular_text_reader_data.n_particles + self.n_header_lines + 1,
             )
             header_1 = file.readline()
-            sample_rate = int(round(_get_time(header_1) - _get_time(header)))
+
+            time_0 = _get_time(header)
+            time_1 = _get_time(header_1)
+            if time_0 is not None and time_1 is not None:
+                sample_rate = int(round(time_1 - time_0))
+            else:
+                sample_rate = None
```

Comment:
Also, looking at it after a few days, there should be some comment here letting the user know that this has not been recorded as it is very important for any dynamic quantities being computed.

Reply:
[b8bb054](https://github.com/zincware/MDSuite/pull/501/commits/b8bb0541607989518264fb0d52aa4efe241c976c)